sdk/test-utils/test-utils/src/tracing/testInstrumenter.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  Instrumenter,
+  InstrumenterSpanOptions,
+  TracingContext,
+  TracingSpan,
+  TracingSpanContext
+} from "@azure/core-tracing";
+import { ContextImpl } from "./contextImpl";
+import { TestTracingSpan } from "./testTracingSpan";
+
+export class TestInstrumenter implements Instrumenter {
+  public contexts: TracingContext[] = [new ContextImpl()];
```

Comment:
We need some ways to assert:
1. All spans are closed (I think that's already supported)
2. There is only one root span (so we should be able to get the root spans)
3. Expected span graph? Like, can I get all the children of a span and assert that their names are what I expect?